src/ContentScript/injectors/youtube.plugin.ts

```diff
@@ -1,10 +1,67 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable no-unused-vars */
+// import { debounce } from 'lodash';
+import LikeButton from '../../sdk/button.iframe';
+
 class YoutubePlugin {
+  youtubeStyle!: HTMLElement;
+
   constructor() {
     console.log('workd');
   }
 
+  insertStyle() {
+    this.youtubeStyle = document.createElement('style');
+    this.youtubeStyle.innerHTML = `
+              #meta-contents {
+                display: flex;
+                flex-direction: row;
+                justify-content: space-between;
+              }
+          `;
+    document.body.appendChild(this.youtubeStyle);
+  }
+
   inject() {
     console.log(this);
+    this.onPageLoaded();
+  }
+
+  private onPageLoaded() {
+    if (!document.querySelector('ytd-app')) {
+      setTimeout(() => {
+        this.onPageLoaded();
+      }, 500);
+    } else {
+      const id = this.getLikeId() || 'foundation';
+      this.insertLikeCoinButton(id);
+    }
+  }
+
+  private getLikeId() {
+    const desc = document.querySelector('#description');
+    console.log(desc, this);
+    const nodes = desc?.querySelectorAll('a') as NodeListOf<HTMLAnchorElement>;
+    console.log(nodes);
+    if (nodes.length === 0) return;
+    const node = nodes[nodes.length - 1];
+    const url = node.innerText;
```

Comment:
What kind of magic url are we expecting here? Maybe we should check if the url contains `like.co` or `button.like.co` and use regex below for robustness?